### PR TITLE
chore: add authentication to docker container

### PR DESCRIPTION
**Pull request by damieva**

### CODE REVIEW

File: pkg/uc/dependencymanager/container_test_types.go [hunk 1]
```diff
@@ -21,4 +21,5 @@ type ContainerRunConfig struct {
 	ExtraHosts   []string                             `yaml:"extra_hosts"`
 	Labels       map[string]string                    `yaml:"labels"`
 	PortBindings map[docker.Port][]docker.PortBinding `yaml:"port_bindings"`
+	Auth         docker.AuthConfiguration             `json:"dockerauth"`
```
Comment (hrodes) [inline pkg/uc/dependencymanager/container_test_types.go:24]: yaml: instead of json

**Review by hrodes — CHANGES_REQUESTED** [review]

File: pkg/uc/dependencymanager/container_test_types.go (hunk 1 above)
Comment (damieva) [reply pkg/uc/dependencymanager/container_test_types.go:24]: Sure, I didn't push the change because I'm testing in local.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.1.5] - 2023-02-27
+
+### Added
+
+- Injection of container registry credentials. They are included as an optional parameter in `ContainerRunConfig` struct.
```
Comment (fllaca) [inline CHANGELOG.md:13]:
```suggestion
- Injection of container registry credentials. They are included as an optional parameter `dockerauth` in `container` [dependency config](https://github.com/Adhara-Tech/itrunner#dependencies-configuration) section.
```

**Review by fllaca — COMMENTED** [review]
LGTM! Just a little suggestion to make the changelog entry a bit more "user oriented". The `ContainerRunConfig` struct is an internal implementation aspect that they, as users of this library, don't see.

**Review by fllaca — APPROVED** [review]

**Pull request merged**